### PR TITLE
Add jwks topic and transport

### PR DESCRIPTION
**Pull request by yalosev**

### CODE REVIEW

File: vault-plugins/flant_iam/backend/factory.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/flant/negentropy/vault-plugins/flant_iam/model"
 	sharedio "github.com/flant/negentropy/vault-plugins/shared/io"
 	sharedkafka "github.com/flant/negentropy/vault-plugins/shared/kafka"
+	shareddest "github.com/flant/negentropy/vault-plugins/shared/kafka/destination"
```
Comment (name212) [inline vault-plugins/flant_iam/backend/factory.go:17]: May be in jwt module?

File: vault-plugins/shared/jwt/jwks_model.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+					},
+				},
+			},
+		},
+	}
+}
+
+type JWKS struct {
+	UUID   string `json:"uuid"`
+	Pubkey *ed25519.PublicKey
```
Comment (name212) [inline vault-plugins/shared/jwt/jwks_model.go:34]: pudkey will not marshal and not send in replicas. Do I right?

File: vault-plugins/shared/kafka/source/jwks.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+	return &JWKSKafkaSource{
+		kf: kf,
+
+		stopC:  make(chan struct{}),
+		logger: logger,
+	}
+}
+
+func (rk *JWKSKafkaSource) Name() string {
+	return "jwks"
```
Comment (name212) [inline vault-plugins/shared/kafka/source/jwks.go:40]: Do we make "jwks" constant?

File: vault-plugins/shared/kafka/source/jwks.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	rk.logger.Debug("Restore - got restoration reader")
+
+	return sharedkafka.RunRestorationLoop(r, runConsumer, "jwks", txn, rk.restoreMsgHandler)
+}
+
+func (rk *JWKSKafkaSource) restoreMsgHandler(txn *memdb.Txn, msg *kafka.Message) error {
+	rk.logger.Debug("Restore - handler run")
+	splitted := strings.Split(string(msg.Key), "/")
+	if len(splitted) != 2 {
+		log.Printf("wrong object Key format: %s\n", msg.Key)
```
Comment (name212) [inline vault-plugins/shared/kafka/source/jwks.go:59]: `rk.logger` need here

File: vault-plugins/shared/kafka/source/jwks.go [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+		if len(signature) == 0 {
+			log.Printf("no signature found. Skipping message: %s in topic: %s at offset %d\n",
+				msg.Key, *msg.TopicPartition.Topic, msg.TopicPartition.Offset)
+			return
+		}
+
+		err := rk.verifySign(signature, msg.Value)
+		if err != nil {
+			log.Printf("wrong signature. Skipping message: %s in topic: %s at offset %d\n",
```
Comment (name212) [inline vault-plugins/shared/kafka/source/jwks.go:135]: and here

**Review by name212 — APPROVED** [review]

File: vault-plugins/shared/jwt/jwks_model.go (hunk 2 above)
Comment (yalosev) [reply vault-plugins/shared/jwt/jwks_model.go:34]: Nope :) It will

**Pull request merged**